### ISSUE TITLE
Error: has incompatible types: sql.NullString, interface{}

### ISSUE THREAD

**Issue by omid9h** [sqlc-dev/sqlc#3656]
I have a MySQL query with multiple named parameters. everything is fine but for bellow chunk:
```sql
AND (
    sqlc.narg(phone_filter) IS NULL
    OR (
        (
            length (sqlc.narg(phone_filter)) != 11
            OR aes_decrypt (from_base64 (u.phone), sqlc.arg(aes_key)) = sqlc.narg(phone_filter)
        )
        AND (
            length (sqlc.narg(phone_filter)) = 11
            OR aes_decrypt (from_base64 (u.phone), sqlc.arg(aes_key)) LIKE '%' || sqlc.narg(phone_filter) || '%'
        )
    )
)
```
and for the last line `OR aes_decrypt (from_base64 (u.phone), sqlc.arg(aes_key)) LIKE '%' || sqlc.narg(phone_filter) || '%'` 

I get this error:

`# package demorepo
error generating code: named param PhoneFilter has incompatible types: sql.NullString, interface{}`

any idea how to fix this? thanks

**Comment by omid9h** [comment]
Something that I've found is that sqlc assumes the parameter here at `LIKE '%' || sqlc.narg(phone_filter) || '%'` to be interface{} and apparently it conflicts with `sql.NullString` which is assumed in above lines.

So, I was able to pass the error by using a second param like `sqlc.narg(phone_filter2)` but now my question is why this syntax `sqlc.narg(phone_filter2)::varchar/char/text/or whatever` does not work as the document mentioned. and in general how can I use casting in MySQL.

**Comment by omid9h** [comment]
solved by not using `sqlc.narg` or `sqlc.arg` in `LIKE` statement.

**Comment by bingbeann** [comment]
@omid9h have you found a solution on this other than using another named parameter? FYI `::<type>` is only supported for PostgreSQL, I believe.

**Comment by omid9h** [comment]
@bingbeann  nah. I just did the work around I mentioned and call it the day. and yes you're right my most use cases is on postgres and there is no problem for it.
